features/topology-map/plugins/org.opennms.features.topology.plugins.topo.graphml/src/main/java/org/opennms/features/topology/plugins/topo/graphml/GraphMLTopologyProvider.java

```diff
@@ -200,7 +214,26 @@ public boolean contributesTo(ContentType type) {
         return Sets.newHashSet(ContentType.Alarm, ContentType.Node).contains(type);
     }
 
-    public boolean requiresStatusProvider() {
-        return requiresStatusProvider;
+    public VertexStatusProviderType vertexStatusProvider() {
+        if (this.vertexStatusProvider instanceof Boolean) {
+            if (this.vertexStatusProvider == Boolean.TRUE) {
+                return VertexStatusProviderType.DEFAULT_STATUS_PROVIDER;
+            } else {
+                return VertexStatusProviderType.NO_STATUS_PROVIDER;
+            }
+
+        } else if (this.vertexStatusProvider instanceof String) {
+            if ("default".equalsIgnoreCase((String)this.vertexStatusProvider)) {
+                return VertexStatusProviderType.DEFAULT_STATUS_PROVIDER;
+            }
+            if ("script".equalsIgnoreCase((String)this.vertexStatusProvider)) {
+                return VertexStatusProviderType.SCRIPT_STATUS_PROVIDER;
+            }
+            if ("propagate".equalsIgnoreCase((String)this.vertexStatusProvider)) {
+                return VertexStatusProviderType.PROPAGATE_STATUS_PROVIDER;
+            }
+        }
+
+        return VertexStatusProviderType.NO_STATUS_PROVIDER;
```

Comment:
In general this should be set while initializing the graph provider to be consistent with the current implementation. See constructor for szl, etc.